### PR TITLE
H-2480: Convert browser plugin to use Flows

### PR DESCRIPTION
**Pull request by CiaranMn**
## 🌟 What is the purpose of this PR?

This PR converts the browser plugin to use flows instead of its legacy code, mainly:
1. Introducing two special flow definitions for manual and automatic browser plugin inference
2. Updating the browser plugin and API to trigger these flows instead of the previous dedicated workflow for browser inference
3. Replace the old 'Logs' tab with a 'History' tab to show flows (internal [design](https://www.figma.com/design/gydVGka9FjNEg9E2STwhi2/HASH?node-id=11668-63362&t=nRuY5RyEkaTIiie7-0))
4. Delete the code the old browser plugin inference functions were using, apart from anything that is still in use by other flow actions (to be cleaned up later)

Some changes to the Temporal workfow
1. Throw errors from the workflow rather than returning them so Temporal views the workflow as failed
2. Switch from using `WAIT_CANCELLATION_COMPLETED` for activities because activities are not set up to receive cancellation signals and clean up (H-2545)
3. Set explicit activity ids on the non-Flow step activities (persist flow, check user permission) to avoid auto-generated ids clashing with our step ids

Tangentially:
1. Add a `getFlowRunById` resolver and refactor the existing `getFlowRuns` to allow for requesting a list of flow runs without needing to check the event history if it isn't required – I ended up not needing this but I will use it for H-1323 so that we can poll for lists of flows without bothering with the event history for all of them.
3. Update `getFlowRuns` to fetch the FlowRun entities visible to the user instead of fetching all flows in Temporal

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] are internal and do not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] do not affect the execution graph

## ⚠️ Known issues

- H-1323: Some pages the plugin links to don't yet exist (up next), e.g. a page listing all flows
- H-2715: The history tab in the plugin doesn't show background requests from flows for pages
- H-2669: Overflow if the DAG is too wide for the screen isn't handled properly if the nav sidebar is open

## 🐾 Next steps

H-1323.

## 🛡 What tests cover this?

None yet

## ❓ How to test this?

1. Run the API, AI worker and plugin locally, load plugin into browser
6. Trigger some manual flows from the plugin or configure automatic and browse
7. Check history tab, check view of flow definitions + runs in HASH

## 📹 Demo

https://github.com/hashintel/hash/assets/37743469/dfee89d6-c7bc-4183-a42c-530e74623ca7

### CODE REVIEW

**Review by benwerner01 — APPROVED** [review]
Thank you Ciaran, the code looks good to me. Had a couple of thoughts, but nothing that should block merging.

Will test out the functionality in the PR on top of the stack, as we agreed after standup.

File: libs/@local/hash-isomorphic-utils/src/flows/browser-plugin-flow-definitions.ts [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+        {
+          inputName:
+            "webPage" satisfies InputNameForAction<"processAutomaticBrowsingSettings">,
+          kind: "step-output",
+          sourceStepId: "trigger",
+          sourceStepOutputName:
+            "visitedWebPage" satisfies AutomaticInferenceTriggerInputName,
+        },
+      ],
+    },
```
[the remark this answers is not in the document]
Comment (benwerner01) [reply libs/@local/hash-isomorphic-utils/src/flows/browser-plugin-flow-definitions.ts:162]: We discussed this internally, and it makes more sense to me now why this was done although I think we may want to revisit this when we have the ability to persist flow definitions, and define a flow definition per web page where the user wants some automated functionality to occur.

File: libs/@local/hash-isomorphic-utils/src/flows/browser-plugin-flow-definitions.ts [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+  ManualInferenceTriggerInputName,
+  ManualInferenceTriggerInputs,
+} from "@local/hash-isomorphic-utils/flows/browser-plugin-flow-types";
+import { browserInferenceFlowOutput } from "@local/hash-isomorphic-utils/flows/browser-plugin-flow-types";
+import type { FlowDefinition } from "@local/hash-isomorphic-utils/flows/types";
+import type { EntityUuid } from "@local/hash-subgraph";
+
+export const manualBrowserInferenceFlowDefinition: FlowDefinition = {
+  name: "Research triggered manually from browser",
+  flowDefinitionId: "manual-browser-inference" as EntityUuid,
```
[the remark this answers is not in the document]
Comment (CiaranMn) [reply libs/@local/hash-isomorphic-utils/src/flows/browser-plugin-flow-definitions.ts:17]: Yes, agreed

File: apps/hash-frontend/src/pages/[shortname]/flows/[flow-uuid].page/flow-definition.tsx [hunk 3]
```diff
@@ -345,42 +354,51 @@ export const FlowDefinition = () => {
 
   const flowDefinitionStateKey = `${selectedFlow.name}`;
   const flowRunStateKey = `${flowDefinitionStateKey}-${
-    selectedFlowRun?.workflowId ?? "definition"
+    selectedFlowRun?.flowRunId ?? "definition"
   }`;
 
+  const isRunnableFromHere = !unrunnableDefinitionIds.includes(
+    selectedFlow.flowDefinitionId,
+  );
```
[the remark this answers is not in the document]
Comment (CiaranMn) [reply apps/hash-frontend/src/pages/[shortname]/flows/[flow-uuid].page/flow-definition.tsx:362]: Yes, good idea. I'll add this in a PR further up the stack so I can start getting them merged

**Pull request merged**